core/coreunix/add.go

```diff
@@ -67,42 +67,46 @@ type AddedObject struct {
 	Bytes int64  `json:",omitempty"`
 }
 
-func NewAdder(ctx context.Context, n *core.IpfsNode, out chan interface{}) (*Adder, error) {
-	mr, err := mfs.NewRoot(ctx, n.DAG, newDirNode(), nil)
+func NewAdder(ctx context.Context, p pin.Pinner, bs bstore.GCBlockstore, ds dag.DAGService) (*Adder, error) {
```

Comment:
curious why the interface change? 

I imagine that this is nicer for users:

```go
node := mkipfsNode(...)
add.NewAdder(ctx, node, out)
```
Was there a case where you didnt have a node, but you had everything else?